languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -34110,6 +34110,125 @@ though, and has since been slightly modified)-->
+              <example correction="Pourtant,"><marker>Pourtant</marker> manger entre les repas est nocif pour la santé.</example>
+              <example correction="Néanmoins,"><marker>Néanmoins</marker> Marie n'en pensait pas moins.</example>
+              <example correction="Toutefois,"><marker>Toutefois</marker> le maire était d'humeur généreuse aujourd'hui.</example>
+          </rule>
+          <rule>
+              <antipattern>
+                  <token>en</token>
+                  <token>conséquence</token>
+                  <token regexp="yes">de|du|des|directe|et</token>
+              </antipattern>
+              <pattern>
+                  <token postag="SENT_START"/>
+                  <token>en</token>
+                  <marker>
+                      <token regexp="yes">effet|conséquence</token>
+                  </marker>
+                  <token postag="[RAVNJPZCD].*" postag_regexp="yes"/>
+              </pattern>
+              <message>Une virgule semble être requise dans ce contexte.</message>
+              <suggestion>\3,</suggestion>
+              <example correction="effet,">En <marker>effet</marker> il ne le remarqua pas.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <token postag="SENT_START"/>
+                  <token>par</token>
+                  <marker>
+                      <token regexp="yes">conséquent|contre</token>
+                  </marker>
+                  <token postag="[RAVNJPZCD].*" postag_regexp="yes"/>
+              </pattern>
+              <message>Une virgule semble être requise dans ce contexte.</message>
+              <suggestion>\3,</suggestion>
+              <example correction="contre,">Par <marker>contre</marker> il ne le remarqua pas.</example>
+          </rule>
+          <rule>
+              <antipattern>
+                  <token regexp="yes">moi|toi</token>
+                  <token regexp="yes">aussi|non|et|qui|absolu|ou</token>
+              </antipattern>
+              <antipattern>
+                  <token regexp="yes">moi|toi</token>
+                  <token>d'</token>
+                  <token>abord</token>
+              </antipattern>
+              <pattern>
+                  <token postag="SENT_START"/>
+                  <marker>
+                      <token regexp="yes">moi|toi</token>
+                  </marker>
+                  <token postag="[RAVNJPZCD].*" postag_regexp="yes"/>
+              </pattern>
+              <message>Une virgule semble être requise dans ce contexte.</message>
+              <suggestion>\2,</suggestion>
+              <example correction="Moi,"><marker>Moi</marker> je ne le sais pas.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <token postag="SENT_START"/>
+                  <marker>
+                      <token regexp="yes">lui|elle|eux|nous|vous|elles</token>
```

Comment:
Do you think there might be sentences in which the second token does not match the first one in number, person, and gender?

"Lui, il va", "Elle, elle va" → this is probable, I think
"Lui, elle aime", "Nous, vous aimez" → is this realistic??

If the answer is "no", you could unify (with the three above-mentioned features) the 2nd and 3rd tokens.

Reply:
Indeed, these exemples are not realistic but unification would exclude sentences like "Nous, on aime la musique classique." or "Lui, je ne veux plus le voir !"